gremlin-go/driver/graphBinary.go

```diff
@@ -199,6 +201,90 @@ func mapReader(buffer *bytes.Buffer, typeSerializer *graphBinaryTypeSerializer)
 	return valMap, nil
 }
 
+func getSignedBytesFromBigInt(n *big.Int) []byte {
+	var one = big.NewInt(1)
+	switch n.Sign() {
+	case 0:
+		return []byte{}
+	case 1:
+		b := n.Bytes()
+		if b[0]&0x80 > 0 {
+			b = append([]byte{0}, b...)
+		}
+		return b
+	case -1:
+		length := uint(n.BitLen()/8+1) * 8
+		b := new(big.Int).Add(n, new(big.Int).Lsh(one, length)).Bytes()
+		if len(b) >= 2 && b[0] == 0xff && b[1]&0x80 != 0 {
+			b = b[1:]
+		}
+		return b
+	}
+	panic("unreachable")
+}
+
+func getBigIntFromSignedBytes(b []byte) *big.Int {
+	var newBigInt = big.NewInt(0).SetBytes(b)
+	var one = big.NewInt(1)
+	if len(b) == 0 {
+		return newBigInt
+	}
+	switch b[0] & 0x80 {
+	case 0x00:
```

Comment:
Same suggestion as above, change flow control such that panic is not required and also add a comment or two

Reply:
here we only have two cases, either b[0] & 0x80 > 0 or == 0, so we can remove the extra case of entering panic.